python/ccf/clients.py

```diff
@@ -36,31 +38,37 @@ def truncate(string, max_len=256):
 
 DEFAULT_CONNECTION_TIMEOUT_SEC = 3
 DEFAULT_REQUEST_TIMEOUT_SEC = 3
+DEFAULT_COMMIT_TIMEOUT_SEC = 3
 
 
 class Request:
-    def __init__(
-        self, path, params=None, http_verb="POST", headers=None, params_in_query=None
-    ):
+    def __init__(self, path, params=None, http_verb="POST", headers=None):
         if headers is None:
             headers = {}
 
-        # TODO: remove
-        if params_in_query is None:
-            params_in_query = http_verb == "GET"
-
         self.path = path
         self.params = params
         self.http_verb = http_verb
         self.headers = headers
-        self.params_in_query = params_in_query
+        self.params_in_query = http_verb == "GET" or http_verb == "DELETE"
+
+    def get_params(self):
```

Comment:
I think we should remove all mention of params, and just have an optional body. Users can add params at the end of the path, there's no need for CCFClient to be aware of the details of the URL string.

Reply:
This is a good approach. We still have this [awkward encoding requirement](https://github.com/microsoft/CCF/pull/1429#discussion_r459456063), but that only really makes a difference where we have string query params (which I think is only `/api`?), so not onerous to do at the calling point.

Reply:
Yes, but that's a quirk we're hoping to iron out eventually, not something we want the CCFClient interface to reflect or know about.

Reply:
Addressed in #1429